### PR TITLE
chore: handle setResourcePool message in commands

### PR DESCRIPTION
**Pull request by eecsliu**
## Description
Change the command behavior so instead of returning an error for unrecognized message, we will respond with an unsupported action error.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/command/command.go [hunk 1]
```diff
@@ -321,12 +321,22 @@ func (c *command) Receive(ctx *actor.Context) error {
 			ctx.Respond(err)
 		}
 
+	case job.SetResourcePool:
+		err := c.setResourcePool(ctx, msg)
```
Comment (hamidzr) [inline master/internal/command/command.go:325]: nit: could be one line.

File: master/internal/command/command.go [hunk 2]
```diff
@@ -321,12 +321,22 @@ func (c *command) Receive(ctx *actor.Context) error {
...
+			ctx.Respond(err)
+		}
+
 	default:
 		return actor.ErrUnexpectedMessage(ctx)
 	}
 	return nil
 }
 
+func (c *command) setResourcePool(ctx *actor.Context, msg job.SetResourcePool) error {
```
Comment (hamidzr) [inline master/internal/command/command.go:336]: if we're going with a sep fn like this. how about having the arg be the resourcepool name?

File: master/internal/command/command.go [hunk 3]
```diff
@@ -321,12 +321,22 @@ func (c *command) Receive(ctx *actor.Context) error {
...
+	return fmt.Errorf("setting resource pool for job type %s is not supported", c.jobType)
```
Comment (hamidzr) [inline master/internal/command/command.go:337]: IMO it'd be nice to include the why in a comment in code.

**Review by hamidzr — APPROVED** [review]

File: master/internal/command/command.go (hunk 2 above)
Comment (eecsliu) [reply master/internal/command/command.go:336]: sure that makes sense. This function is mostly to set up future implementation.

**Pull request merged**